frontend/src/metabase/search/components/SearchResult.unit.spec.tsx

```diff
@@ -0,0 +1,131 @@
+import { render, screen } from "@testing-library/react";
+import { setupEnterpriseTest } from "__support__/enterprise";
+import { createMockSearchResult } from "metabase-types/api/mocks";
+import { getIcon } from "__support__/ui";
+
+import type { WrappedResult } from "./types";
+import { SearchResult } from "./SearchResult";
+
+const createWrappedSearchResult = (
+  options: Partial<WrappedResult>,
+): WrappedResult => {
+  const result = createMockSearchResult(options);
+
+  return {
+    ...result,
+    getUrl: options.getUrl ?? (() => "/collection/root"),
+    getIcon: options.getIcon ?? (() => ({ name: "folder" })),
+    getCollection: options.getCollection ?? (() => result.collection),
+  };
+};
+
+describe("SearchResult", () => {
+  it("renders a search result question item", () => {
+    const result = createWrappedSearchResult({
+      name: "My Item",
+      model: "card",
+      description: "My Item Description",
+      getIcon: () => ({ name: "table" }),
+    });
+
+    render(<SearchResult result={result} />);
+
+    expect(screen.getByText(result.name)).toBeInTheDocument();
+    expect(screen.getByText(result.description as string)).toBeInTheDocument();
+    expect(getIcon("table")).toBeInTheDocument();
+  });
+
+  it("renders a search result collection item", () => {
+    const result = createWrappedSearchResult({
+      name: "My Folder of Goodies",
+      model: "collection",
+      collection: {
+        id: 1,
+        name: "This should not appear",
+        authority_level: null,
+      },
+    });
+
+    render(<SearchResult result={result} />);
+
+    expect(screen.getByText(result.name)).toBeInTheDocument();
+    expect(screen.getByText("Collection")).toBeInTheDocument();
+    expect(screen.queryByText(result.collection.name)).not.toBeInTheDocument();
+    expect(getIcon("folder")).toBeInTheDocument();
+  });
+});
+
+describe("SearchResult > Collections", () => {
+  const resultInRegularCollection = createWrappedSearchResult({
+    name: "My Regular Item",
+    collection_authority_level: null,
+    collection: {
+      id: 1,
+      name: "Regular Collection",
+      authority_level: null,
+    },
+  });
+
+  const resultInOfficalCollection = createWrappedSearchResult({
+    name: "My Official Item",
+    collection_authority_level: "official",
+    collection: {
+      id: 1,
+      name: "Official Collection",
+      authority_level: "official",
+    },
+  });
+
+  describe("OSS", () => {
+    it("renders regular collection correctly", () => {
+      render(<SearchResult result={resultInRegularCollection} />);
+      expect(
+        screen.getByText(resultInRegularCollection.name),
+      ).toBeInTheDocument();
+      expect(screen.getByText("Regular Collection")).toBeInTheDocument();
+      expect(screen.getByLabelText("folder icon")).toBeInTheDocument();
```

Comment:
```suggestion
      expect(getIcon("folder")).toBeInTheDocument();
```
I think we can use `getIcon` and `queryIcon` for all of these test cases.

Reply:
ooo, I didn't know about `queryIcon`